test/highlight/set.vim

```diff
@@ -0,0 +1,10 @@
+set hls!
```

Comment:
Could you check that the `!` is highlighted here too?
Thanks!

Reply:
I haven't highlighted the `!` for consistency with `:set-?` and `:set-&`, but I guess I could add highlight those with an `@operator` capture.

Reply:
Yeah, `operator` seems consistent.

Reply:
I have since made some additional changes to the highlighting (and the tests), can you check if these changes make sense?

Reply:
It looks great !